[DOC] update notes in download page for the decompressing gzip issue [skip ci] (#11400)
NVIDIA/spark-rapids: docs/download.md

* Improve UCX shuffle 
* For updates on RAPIDS Accelerator Tools, please visit [this link](https://github.com/NVIDIA/spark-rapids-tools/releases)

> Note: There is a known issue for 24.08.1 release while decompressing gzip files on H100 GPU.

Reviewer: ```suggestion
Note: There is a known issue in the 24.08.1 release when decompressing gzip files on H100 GPUs.  
```
